discovery/ionos/server_test.go

```diff
@@ -53,63 +54,98 @@ func TestIONOSServerRefresh(t *testing.T) {
-			"__meta_ionos_server_type":              "ENTERPRISE",
-		},
-		{
-			"__address__":                           "85.215.248.84:80",
-			"__meta_ionos_server_availability_zone": "ZONE_1",
-			"__meta_ionos_server_boot_cdrom_id":     "0e4d57f9-cd78-11e9-b88c-525400f64d8d",
-			"__meta_ionos_server_cpu_family":        "INTEL_SKYLAKE",
-			"__meta_ionos_server_id":                "523415e6-ff8c-4dc0-86d3-09c256039b30",
-			"__meta_ionos_server_ip":                ",85.215.248.84,",
-			"__meta_ionos_server_nic_ip_unnamed":    ",85.215.248.84,",
-			"__meta_ionos_server_lifecycle":         "AVAILABLE",
-			"__meta_ionos_server_name":              "prometheus-1",
-			"__meta_ionos_server_servers_id":        "8feda53f-15f0-447f-badf-ebe32dad2fc0/servers",
-			"__meta_ionos_server_state":             "RUNNING",
-			"__meta_ionos_server_type":              "ENTERPRISE",
-		},
-	} {
+	
+	file, err := os.Open("testdata/targets.json")
+	require.NoError(t, err)
+	defer file.Close()
+
+	var jsonTargets []model.LabelSet
+
+	err = json.NewDecoder(file).Decode(&jsonTargets)
+	require.NoError(t, err)
+
+	require.Len(t, tg.Targets, len(jsonTargets))
+
+	for i, lbls := range jsonTargets {
 		t.Run(fmt.Sprintf("item %d", i), func(t *testing.T) {
 			require.Equal(t, lbls, tg.Targets[i])
 		})
 	}
 }
 
 func mockIONOSServers(w http.ResponseWriter, r *http.Request) {
-	if r.Header.Get("Authorization") != fmt.Sprintf("Bearer %s", ionosTestBearerToken) {
-		http.Error(w, "bad token", http.StatusUnauthorized)
-		return
-	}
-	if r.URL.Path != fmt.Sprintf("%s/datacenters/%s/servers", ionoscloud.DefaultIonosBasePath, ionosTestDatacenterID) {
-		http.Error(w, fmt.Sprintf("bad url: %s", r.URL.Path), http.StatusNotFound)
-		return
-	}
-	w.Header().Set("Content-Type", "application/json")
-	server, err := os.ReadFile("testdata/servers.json")
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
-	}
-	_, err = w.Write(server)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
-	}
+    if r.Header.Get("Authorization") != fmt.Sprintf("Bearer %s", ionosTestBearerToken) {
+        http.Error(w, "bad token", http.StatusUnauthorized)
+        return
+    }
+
+    if r.URL.Path != fmt.Sprintf("%s/datacenters/%s/servers", ionoscloud.DefaultIonosBasePath, ionosTestDatacenterID) {
+        http.Error(w, fmt.Sprintf("bad url: %s", r.URL.Path), http.StatusNotFound)
+        return
+    }
+
+    limitStr := r.URL.Query().Get("limit")
+    offsetStr := r.URL.Query().Get("offset")
+    
+    limit := 1000 
+    offset := 0   
+
+    if limitStr != "" {
+        if l, err := strconv.Atoi(limitStr); err == nil {
+            limit = l
+        }
+    }
+
+    if offsetStr != "" {
+        if o, err := strconv.Atoi(offsetStr); err == nil {
+            offset = o
+        }
+    }
+
+    w.Header().Set("Content-Type", "application/json")
+    
+    serverData, err := os.ReadFile("testdata/servers.json")
```

Comment:
Does the testdata really need to be this big or can we use a more reasonably sized version? 
Haven't checked it out in my IDE yet so could you explain what's being tested exactly?